NimbleMedium/Sources/Presentation/Modules/Login/LoginViewModel.swift

```diff
@@ -26,11 +31,45 @@ protocol LoginViewModelProtocol: ObservableViewModel {
 
 final class LoginViewModel: ObservableObject, LoginViewModelProtocol {
 
+    private let disposeBag = DisposeBag()
+    private let loginUseCase: LoginUseCaseProtocol
+
     var input: LoginViewModelInput { self }
     var output: LoginViewModelOutput { self }
 
+    @PublishRelayProperty var didLogin: Signal<Void>
+    @PublishRelayProperty var didSelectNoAccount: Signal<Void>
+    @PublishRelayProperty var errorMessage: Signal<String>
+    @BehaviorRelayProperty(false) var isLoading: Driver<Bool>
+
+    init(factory: ModuleFactoryProtocol) {
+        loginUseCase = factory.loginUseCase()
+    }
+
+    private func login(email: String, password: String) {
+        loginUseCase
+            .login(email: email, password: password)
+            .subscribe(onCompleted: { [weak self] in
+                self?.$isLoading.accept(false)
+                self?.$didLogin.accept(())
+            }, onError: { [weak self] in
+                self?.$isLoading.accept(false)
+                self?.$errorMessage.accept($0.detail)
+            })
```

Comment:
We can use `subscribe(with: self)`
```suggestion
            .subscribe(
                with: self,
                onCompleted: { owner in
					owner.$isLoading.accept(false)
	                owner.$didLogin.accept(())
                },
                onError: { owner, error in
					owner.$isLoading.accept(false)
	                owner.$errorMessage.accept($0.detail)
                }
            )
```